src/main/java/org/jenkinsci/plugins/ownership/integrations/securityinspector/OwnerFilter.java

```diff
@@ -0,0 +1,158 @@
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.servlet.ServletException;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.ownership.model.OwnershipHelperLocator;
+import org.jenkinsci.plugins.securityinspector.util.JenkinsHelper;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.StaplerRequest;
+
+public class OwnerFilter {
+    
+    /**
+     * Include regex string.
+     */
+    @CheckForNull
+    private final String includeRegex;
+
+    /**
+     * Compiled include pattern from the includeRegex string.
+     */
+    @CheckForNull
+    private transient Pattern includePattern;
+    
+    /**
+     * Constructs empty filter.
+     */
+    public OwnerFilter() {
+        this.includeRegex = null;
+        this.includePattern = null;
+    }
+    
+    @Restricted(NoExternalUse.class)
+    public OwnerFilter(@Nonnull StaplerRequest req)
+            throws Descriptor.FormException, ServletException {
+        if (req.getParameter("useincluderegex") != null) {
+            includeRegex = Util.nullify(req.getParameter("_.includeRegex"));
+            if (includeRegex == null) {
+                includePattern = null;
+            } else {
+                includePattern = Pattern.compile(includeRegex);
+            }
+        } else {
+            includeRegex = null;
+            includePattern = null;
+        }
+    }
+    
+    @Nonnull
+    @Restricted(NoExternalUse.class)
+    public List<TopLevelItem> doFilter(User owner) {
+        
+        final SortedSet<String> names = new TreeSet<>();
+        
+        final Jenkins jenkins = JenkinsHelper.getInstanceOrFail();
+        final List<Item> allItems = jenkins.getAllItems(Item.class);
+        String itemName;
+        
+        for (Item item : allItems) {
+            AbstractOwnershipHelper<Item> located = OwnershipHelperLocator.locate(item);
+            if (located == null) {
+                continue;
+            }
+            
+            OwnershipDescription ownershipDescription = located.getOwnershipDescription(item);
+            if (ownershipDescription.isOwner(owner, true)) {
+                itemName = item.getFullName();
+                names.add(itemName);
+            }
+        }
+        
+        if (includePattern != null) {
+            for (Item item : allItems) {
+                AbstractOwnershipHelper<Item> located = OwnershipHelperLocator.locate(item);
+                if (located == null) {
+                    continue;
+                }
+            
+                itemName = item.getFullName();
+                OwnershipDescription ownershipDescription = located.getOwnershipDescription(item);
+                if (ownershipDescription.isOwner(owner, true) 
+                        && includePattern.matcher(itemName).matches()) {
+                    names.add(itemName);
+                }
+            }
+        } else {
```

Comment:
The clause below is mostly a copy-paste. Maybe you could optimize the code somehow to avoid duplications